cpp/src/gandiva/function_registry_math_ops.cc

```diff
@@ -62,11 +73,16 @@ namespace gandiva {
 std::vector<NativeFunction> GetMathOpsFunctionRegistry() {
   static std::vector<NativeFunction> math_fn_registry_ = {
       MATH_UNARY_OPS(cbrt, {}), MATH_UNARY_OPS(exp, {}), MATH_UNARY_OPS(log, {}),
-      MATH_UNARY_OPS(log10, {}),
+      MATH_UNARY_OPS(log10, {}), MATH_UNARY_OPS(sqrt, {}),
+
+      MATH_UNARY_OPS_FLOAT(sqrtf, {}), MATH_UNARY_OPS_FLOAT(cbrtf, {}),
+      MATH_UNARY_OPS_FLOAT(expf, {}), MATH_UNARY_OPS_FLOAT(logf, {}),
+      MATH_UNARY_OPS_FLOAT(log10f, {}),
```

Comment:
Why not overload the existing functions?

Reply:
You mean like:

```suggestion
      MATH_UNARY_OPS_FLOAT(log10, {}),
```

we can do that, we just liked the fact there was no implicit conversion. Do you have strong feelings about that?

Reply:
The reasoning here is that we deal with limited precision data and we want to avoid automatic promotion to double from float, which has memory and performance implications. Having float-only functions allows us to spot the issues where extra precision may be used incorrectly at compilation time.

Reply:
On second thought I think appending an "f" to function names makes sense. Since the new functions differ from existing ones only by their return types but not input types, I agree that they should not have the same names.